src/main/java/com/programmers/springbootbasic/domain/voucher/controller/VoucherController.java

```diff
@@ -0,0 +1,101 @@
+package com.programmers.springbootbasic.domain.voucher.controller;
+
+import com.programmers.springbootbasic.common.response.model.CommonResult;
+import com.programmers.springbootbasic.common.response.model.ListResult;
+import com.programmers.springbootbasic.common.response.service.ResponseFactory;
+import com.programmers.springbootbasic.domain.voucher.dto.VoucherRequestDto;
+import com.programmers.springbootbasic.domain.voucher.entity.Voucher;
+import com.programmers.springbootbasic.domain.voucher.exception.ErrorMsg;
+import com.programmers.springbootbasic.domain.voucher.service.VoucherService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Controller;
+
+import java.util.UUID;
+
+@Slf4j
+@Controller
+@RequiredArgsConstructor
+public class VoucherController {
+    private final VoucherService voucherService;
+
+    public CommonResult createVoucher(String voucherType, String value) {
+        try {
+            voucherService.createVoucher(VoucherRequestDto.builder()
+                    .voucherType(Integer.parseInt(voucherType))
+                    .value(Long.parseLong(value))
+                    .build());
+        } catch (NumberFormatException e) {
+            log.warn(e.toString());
+            return ResponseFactory.getFailResult(ErrorMsg.numberFormatMismatch.getMessage());
+        } catch (Exception e) {
+            log.warn(e.toString());
+            return ResponseFactory.getFailResult(e.getMessage());
+        }
+        return ResponseFactory.getSuccessResult();
+    }
```

Comment:
비슷한 형식의 구문이 계속 반복되가고 있는 것 같습니다.

템플릿 메서드 패턴을 고려해볼 수도 있을 것 같은데 어떻게 생각하시나요?

Reply:
Controller 부분을 작성하면서 반복에 대한 고민이 있었는데 이에 대한 좋은 해답일 것 같습니다!

```suggestion
    public CommonResult<String> createVoucher(String voucherType, String value) {
        return execute(voucherService::createVoucher, VoucherRequestDto.builder()
                .voucherType(Integer.parseInt(voucherType))
                .value(Long.parseLong(value))
                .build());
    }
    
        private CommonResult<String> execute(Consumer<VoucherRequestDto> consumer, VoucherRequestDto voucherRequestDto) {
        try {
            consumer.accept(voucherRequestDto);
            return CommonResult.getSuccessResult();
        } catch (NumberFormatException e) {
            log.warn(e.toString());
            return CommonResult.getFailResult(ErrorMsg.NUMBER_FORMAT_MISMATCH.getMessage());
        } catch (IllegalArgumentException e) {
            log.warn(e.toString());
            return CommonResult.getFailResult(ErrorMsg.UUID_FORMAT_MISMATCH.getMessage());
        } catch (Exception e) {
            log.warn(e.toString());
            return CommonResult.getFailResult(e.getMessage());
        }
    }
```

이러한 방식으로 적용해보겠습니다!